xbmc/FileItem.cpp

```diff
@@ -1706,6 +1706,7 @@ void CFileItem::UpdateInfo(const CFileItem &item, bool replaceLabels /*=true*/)
       m_videoInfoTag = new CVideoInfoTag;
     }
 
+    SetDynPath(item.GetDynPath());
```

Comment:
please also check if this is needed. GetDynpath will fallback to the path if dynpath is not set. I don't see a reason why you need to set a dynpath to the already defined dynpath. The only reason I see is to force the path to be the dynpath in cases where the dynpath is not set. But then, getdynpath will always fallback to the path which makes me question the need here